File: examples/examples_CPU_FORTRAN/gm55.f90

PROGRAM GM55_Example

   PARAMETER (N=100000000)
   INTEGER i,GM55_generate,sum
   REAL val,rsum

   TYPE GM55_state
     INTEGER z(36)
   END TYPE GM55_state
   TYPE(GM55_state) state

   CALL GM55_init ( state )
   CALL GM55_Print_State (state)

   sum = 0

   DO i=1,N
     sum = sum + GM55_generate (state)
   END DO

   WRITE(*,1) N
   rsum = sum/4294967296.
   val = GM55_generate (state)/4294967296.
   IF (rsum<0) THEN
     rsum=rsum+1
   ENDIF
   IF (val<0) THEN
     val=val+1
   ENDIF
   WRITE(*,2) rsum
   WRITE(*,3) val

1  FORMAT(I9," GM55 pseudorandom numbers generated.")
2  FORMAT("Fractional part of the total sum of generated numbers: ",F8.6)
3  FORMAT("Next output value: ",F8.6)

END PROGRAM GM55_Example

File: examples/examples_CPU_C/gm55.c

#include<stdio.h>
#include<gm55.h>

#define NN       100000000UL

int main(void){ 
   long i; unsigned int sum=0;
   gm55_state state;
   gm55_init_(&state);
   gm55_print_state_(&state);
   for(i=0;i<NN;i++) sum+=gm55_generate_(&state);
   printf("%ld GM55 pseudorandom numbers generated.\n",NN);
   printf("Fractional part of the total sum of generated numbers: %f\n",sum/4294967296.);
   printf("Next output value: %f\n",gm55_generate_(&state)/4294967296.);
   return 0;
}
